colcon_core/task/python/build.py

```diff
@@ -244,10 +249,14 @@ def _symlinks_in_build(self, args, setup_py_data):
                 os.path.join(args.build_base, item)))
         # provide a symlink within the build space if a module name is
         # changed by the mapping specified in package_dir
+        temp_symlinks = []
         for rel_src, rel_dst in renamed_items:
             symlinks.append((
-                os.path.join(args.build_base, rel_src),
+                os.path.join(args.path, item),
```

Comment:
this makes the symlink point to source space because the path it's pointing to gets deleted by the other new code in this PR